src/main/java/com/es/phoneshop/model/entity/Product.java

```diff
@@ -44,4 +44,9 @@ public Product(String code, String description, BigDecimal price, Currency curre
         this.imageUrl = imageUrl;
         this.priceHistory = priceHistoryList;
     }
+
+    @Override
+    public int compareTo(Product o) {
```

Comment:
do we really need Comparable<Product> and such compareTo?

Reply:
in order not to return null in comparators, but Comparator.naturalOrdering

Reply:
we shouldn't hack POJO to make DAO working, can this be done in a different way?